homeassistant/components/amcrest/camera.py

```diff
@@ -387,8 +404,10 @@ def update(self) -> None:
                 else:
                     self._model = "unknown"
             if self._attr_unique_id is None:
-                self._attr_unique_id = self._api.serial_number.strip()
-                _LOGGER.debug("Assigned unique_id=%s", self._attr_unique_id)
+                serial_number = self._api.serial_number.strip()
+                if serial_number:
+                    self._attr_unique_id = f"{serial_number}-{self._resolution}"
+                    _LOGGER.debug("Assigned unique_id=%s", self._attr_unique_id)
```

Comment:
```suggestion
                    msg = f"Assigned unique_id={self._attr_unique_id}¨
                    _LOGGER.debug(msg)
```

Reply:
For the Logger, we should use lazy formating and not assign a local constants with templates if we can